### ISSUE TITLE
Intercepted bean method public final Controller.getRequest() cannot be declared final

### ISSUE THREAD

**Issue by skaarr** [decebals/pippo#387]
Now impossible use controller with class annotated cdi interseptor.

`@Secured
@ApplicationScoped
public class FormController
    extends Controller
{
}`

Is there any way to remove `final` keyword from `Controller` public methods?

**Comment by decebals** [comment]
The `Controller` class plays only a decorative role, it contains only helper methods. Read https://github.com/decebals/pippo/pull/350#issuecomment-278413829 for details. I don't know why you want to remove `final` keyword from methods signature.
Maybe all you need is to write an Interceptor. See https://github.com/decebals/pippo-demo/blob/master/pippo-demo-controller/src/main/java/ro/pippo/demo/controller/ContactsController.java#L41 for demo and https://github.com/decebals/pippo/pull/341 if you need more info about controllers.
Read [CDI](http://www.pippo.ro/mod/cdi.html) section from doc if you work with CDI.

**Comment by skaarr** [comment]
Using interceptor causes an exception in my cause:
```
javax.servlet.ServletException: org.jboss.weld.exceptions.DeploymentException: WELD-001504: Intercepted bean method [EnhancedAnnotatedMethodImpl] public final ro.pippo.controller.Controller.getRequest() (intercepted by my.security.SecuredInterceptor) cannot be declared final
	at ro.pippo.core.PippoFilter.init(PippoFilter.java:115)
	at ro.pippo.undertow.websocket.UndertowWebSocketFilter.init(UndertowWebSocketFilter.java:51)
```
annotation example
```
@Inherited
@InterceptorBinding
@Retention(RUNTIME)
@Target({TYPE})
public @interface Secured
{
}
```
interceptor example
```
@Secured
@Interceptor
@Priority(Interceptor.Priority.APPLICATION)
public class SecuredInterceptor
{
  @AroundInvoke
  public Object doSecured(InvocationContext context)
      throws Exception
  {
    return context.proceed();
  }
}
```
So I want to remove final to avoid weld DeploymentException

**Comment by decebals** [comment]
Can you share more code? I want to see some code from controller.
When I said to create an interceptor, I talked about an interceptor from Pippo Controller.
In your case, see if it's possible to exclude some methods from intercepting because you want to intercept only methods from your controller ("route methods") and not the methods inherited from parent (Controller class).

**Comment by decebals** [comment]
https://docs.jboss.org/weld/reference/latest/en-US/html/configure.html#_proxying_classes_with_final_methods

**Comment by skaarr** [comment]
I created a sample application with this exception here https://github.com/skaarr/pippo-cdi-interceptor

Adding weld.properties with org.jboss.weld.proxy.ignoreFinalMethods seems doesn't help (also included in this sample)

**Comment by decebals** [comment]
I didn't have time to take a look. Please clone Pippo on your machine, make the modifications (remove final keywords for the methods of `Controller`), build Pippo on your machine (`mvn clean install`) and if everything is OK in your application, you can submit a PR (Pull Request) or I can push myself the modifications on GitHub.

**Comment by decebals** [comment]
Resolved by #389.
